pkg/recordset/recordset.go

```diff
@@ -178,6 +213,29 @@ func getStacks(cl client.StackDescribeLister, re *regexp.Regexp, installation st
 	return result, nil
 }
 
+func filterStacksByStatus(input []cloudformation.Stack, statuses []string, exclude bool) []cloudformation.Stack {
```

Comment:
In typical functional languages filter functions return values that meet the condition. `exclude` argument makes it all other way around. Let's please replace it with `condition` and flip all `true` and `false` values passed to this function call in the code. Also double negations like `exclude == false` which basically means `include` are hard to reason about. `filterStacksByStatus(sourceStacks, stackStatusCompleteNotDelete, false)` - to me it looks like it returns all stack but those with `stackStatusCompleteNotDelete` which is actually the other way around. I would also argue `(exclude && !match) || (!exclude && match)` is harder to reason about than `condition == match`.

Reply:
I agree, this is kind of confusing. Instead of sticking with this confusing `exclude` argument, I rewrote this in a cleaner way.

`filterStacksByStatus` do filter stacks by status
`excludeStacksByStatus` do exclude stacks by status (does the opposite of `filterStacksByStatus`)
`stackHasStatus` check is stack has status

PTAL